### PR TITLE
Add travis automation to push new releases to PyPI

### PR DESCRIPTION
**Pull request by johngian**

### CODE REVIEW

**Comment by johngian** [comment]
Fixes #143

**Comment by peterbe** [comment]
So what remains here? 

Getting @jvehent_'s input? Adding `on: tags: true` to the `deploy:` block?

Also, I shouldn't merge this PR until we've written all this down in the docs.

**Comment by johngian** [comment]
PR updated with docs

File: README.rst [hunk 1]
```diff
@@ -82,6 +82,21 @@ You can also run linting with ``tox``:
...
 
 
+Releasing a new version
+------------------------
+
+``mozilla-django-oidc`` releases are hosted in `PyPI <https://pypi.python.org/pypi/mozilla-django-oidc>`_.
+Here are the steps you need to follow in order to push a new release:
+
+* Make sure that ``HISTORY.rst`` is up-to-date focusing mostly on backwards incompatible changes
+* Bump the project version and create a commit for that
```
Comment (peterbe) [inline README.rst]: It needs to mention *where*. I believe it's `mozilla_django_oidc/__init__.py` and `setup.cfg` (what is that second file for anyway?)

Comment (peterbe) [inline README.rst]: Oh I get it. There's a tool called "bumpversion" can that does that edit. It's in that requirements_dev.txt.

I tried to use it but perhaps I don't know how:
```
▶ bumpversion 0.3.1
Traceback (most recent call last):
  File "/Users/peterbe/virtualenvs/mozilla-django-oidc/bin/bumpversion", line 11, in <module>
    sys.exit(main())
  File "/Users/peterbe/virtualenvs/mozilla-django-oidc/lib/python3.5/site-packages/bumpversion/__init__.py", line 994, in main
    vcs.commit(message=commit_message)
  File "/Users/peterbe/virtualenvs/mozilla-django-oidc/lib/python3.5/site-packages/bumpversion/__init__.py", line 73, in commit
    list(os.environ.items()) + [(b'HGENCODING', b'utf-8')]
  File "/opt/boxen/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/subprocess.py", line 629, in check_output
    **kwargs).stdout
  File "/opt/boxen/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/subprocess.py", line 711, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'commit', '-F', '/var/folders/1x/2hf5hbs902q54g3bgby5bzt40000gn/T/tmpal20r8pj']' returned non-zero exit status 1
```

File: README.rst [hunk 2]
```diff
@@ -82,6 +82,21 @@ You can also run linting with ``tox``:
...
+   * Run ``bumpversion <patch|minor|major>`` a tool to automate this procedure following the `semantic versioning scheme <http://semver.org/>`_
+* Create a `signed tag <https://git-scm.com/book/tr/v2/Git-Tools-Signing-Your-Work>`_ for that version
```
Comment (peterbe) [inline README.rst]: The command for that would be `git tag -s -a {VERSION} -m "Bump version: {PREVIOUS TAG} {VERSION}"`. Right? Can we add that?

Comment (peterbe) [inline README.rst]: I still like that there's a link to documentation about how to sign tags. This URL is pretty good too: https://help.github.com/articles/signing-tags-using-gpg/

File: README.rst (hunk 1 above)
Comment (johngian) [inline README.rst]: Yeah the call is ``bumpversion patch`` which means i should change the docs :P

File: README.rst [hunk 3]
```diff
@@ -82,6 +82,27 @@ You can also run linting with ``tox``:
     $ tox -e lint
 
 
+Releasing a new version
+------------------------
+
+``mozilla-django-oidc`` releases are hosted in `PyPI <https://pypi.python.org/pypi/mozilla-django-oidc>`_.
+Here are the steps you need to follow in order to push a new release:
+
+* Make sure that ``HISTORY.rst`` is up-to-date focusing mostly on backwards incompatible changes
```
Comment (akatsoulas) [inline README.rst]: Missing a dot `(.)` here.

File: README.rst [hunk 4]
```diff
@@ -82,6 +82,27 @@ You can also run linting with ``tox``:
...
+* Bump the project version and create a commit for the new version
```
Comment (akatsoulas) [inline README.rst]: Either append a `:` or a `.` at the end of line.

**Comment by akatsoulas** [comment]
I am stepping in here after talking with @johngian to move a bit faster due to the time diff so that we can this automation a bit faster.

**Pull request merged**